airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceDatatypeIntegrationTest.kt

```diff
@@ -0,0 +1,254 @@
+            Jsons.createArrayNode().apply { addAll(data.sortedBy { it.toString() }) }
+
+        val actualRecords: List<AirbyteRecordMessage> = actualRead?.records() ?: listOf()
+        val actual: JsonNode = sortedRecordData(actualRecords.mapNotNull { it.data })
+        log.info { "test case $streamName: emitted records $actual" }
+        val expected: JsonNode = sortedRecordData(allStreamNamesAndRecordData[streamName]!!)
+
+        Assertions.assertEquals(expected, actual)
+    }
+
+    companion object {
+        lateinit var dbContainer: MySQLContainer<*>
+
+        fun config(): MysqlSourceConfigurationJsonObject = MysqlContainerFactory.config(dbContainer)
+
+        val connectionFactory: JdbcConnectionFactory by lazy {
+            JdbcConnectionFactory(MysqlSourceConfigurationFactory().make(config()))
+        }
+
+        val stringValues =
+            mapOf(
+                "'abcdef'" to """"abcdef"""",
+                "'ABCD'" to """"ABCD"""",
+                "'OXBEEF'" to """"OXBEEF"""",
+            )
+
+        val testCases: List<TestCase> =
+            listOf(
+                TestCase("VARCHAR(10)", stringValues),
+            )
+
+        val allStreamNamesAndRecordData: Map<String, List<JsonNode>> =
+            testCases.flatMap { it.streamNamesToRecordData.toList() }.toMap()
+
+        @JvmStatic
+        @BeforeAll
+        @Timeout(value = 300)
+        fun startAndProvisionTestContainer() {
+            dbContainer =
+                MysqlContainerFactory.exclusive(
+                    "mysql:8.0",
+                    MysqlContainerFactory.WithNetwork,
+                )
+            connectionFactory
+                .get()
+                .also { it.isReadOnly = false }
+                .use { connection: Connection ->
+                    for (case in testCases) {
+                        for (sql in case.sqlStatements) {
+                            log.info { "test case ${case.id}: executing $sql" }
+                            connection.createStatement().use { stmt -> stmt.execute(sql) }
+                        }
+                    }
+                }
+        }
+    }
+
+    data class TestCase(
+        val sqlType: String,
+        val sqlToAirbyte: Map<String, String>,
+        val airbyteType: AirbyteType = LeafAirbyteType.STRING,
+        val cursor: Boolean = true,
+        val noPK: Boolean = false,
+        val customDDL: List<String>? = null,
+    ) {
+        val id: String
+            get() =
+                sqlType
+                    .replace("[^a-zA-Z0-9]".toRegex(), " ")
+                    .trim()
+                    .replace(" +".toRegex(), "_")
+                    .lowercase()
+
+        val tableName: String
+            get() = "tbl_$id"
+
+        val varrayTableName: String
+            get() = "tbl_varray_$id"
+
+        val columnName: String
+            get() = "col_$id"
+
+        val varrayColumnName: String
+            get() = "col_varray_$id"
```

Comment:
again you can get rid of the varray stuff, that was just for oracle
there's a lot of it in this test; I think you already removed some

Reply:
thanks. Forgot to remove this one.